Editor/AGS.Editor/GUI/GlobalVariableDialog.cs

```diff
@@ -38,25 +38,30 @@ public GlobalVariableDialog(GlobalVariable variable, Game game)
             Utilities.CheckLabelWidthsOnForm(this);
         }
 
-        private void PopulateTypeList()
+        private void PopulateTypeList(IEnumerable<APITypeDef> apiTypes)
         {
             cmbType.Items.Clear();
             cmbType.Items.Add(new GlobalVariableType("int", @"^\-?[0-9]+$"));
             cmbType.Items.Add(new GlobalVariableType("String", @"^.*$"));
             cmbType.Items.Add(new GlobalVariableType("float", @"^\-?[0-9]+(\.[0-9]+)?$"));
             cmbType.Items.Add(new GlobalVariableType("bool", @"^true$|^false$"));
-            cmbType.Items.Add(new GlobalVariableType("GUI*", null));
-            cmbType.Items.Add(new GlobalVariableType("AudioChannel*", null));
-            cmbType.Items.Add(new GlobalVariableType("Character*", null));
-            cmbType.Items.Add(new GlobalVariableType("DynamicSprite*", null));
-            cmbType.Items.Add(new GlobalVariableType("Overlay*", null));
-            cmbType.Items.Add(new GlobalVariableType("ViewFrame*", null));
-            cmbType.SelectedIndex = 0;
+            if (apiTypes != null)
+            {
+                foreach (var def in apiTypes)
+                {
+                    if (!def.Managed || def.String) continue;
+                    if (def.Autoptr)
+                        cmbType.Items.Add(new GlobalVariableType(def.Name, null));
+                    else
+                        cmbType.Items.Add(new GlobalVariableType(string.Format("{0}*", def.Name), null));
+                }
+                cmbType.SelectedIndex = 0;

```

Comment:
Since there will definitely be 4 values added to the control, I would imagine that if `apiTypes` was null for some reason, you would still want to set the SelectedIndex property to 0?

Reply:
Probably misplaced bracket, fixed now.